lnwallet/channel.go

```diff
@@ -4837,29 +4892,103 @@ func (lc *LightningChannel) ReceiveNewCommitment(commitSigs *CommitSigs) error {
 	cancelChan := make(chan struct{})
 	verifyResps := lc.sigPool.SubmitVerifyBatch(verifyJobs, cancelChan)
 
+	localCommitTx := localCommitmentView.txn
+	multiSigScript := lc.signDesc.WitnessScript
+	prevFetcher := txscript.NewCannedPrevOutputFetcher(
```

Comment:
You mean we can _move_ the `hashCache` initialization logic to the `else` case since that's when it's directly used? Yep, the new logic ends up creating a new instance further down the pipeline.